### ISSUE TITLE
[devel] concurrent.futures._base.TimeoutError

### ISSUE THREAD

**Issue by ichorid** [Tribler/tribler#5397]
After merging `release-7.5` into `devel` Windows test for Tunnels fails sometimes:
https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_async_win64/700/artifact/output/test_runners_output/0.err/*view*/

```python
======================================================================
ERROR: Testing whether an anonymous download over our tunnels works
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\site-packages\asynctest\case.py", line 297, in run
    self._run_test_method(testMethod)
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\site-packages\asynctest\case.py", line 354, in _run_test_method
    self.loop.run_until_complete(result)
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\site-packages\asynctest\case.py", line 224, in wrapper
    return method(*args, **kwargs)
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\asyncio\base_events.py", line 579, in run_until_complete
    return future.result()
  File "C:\workspace\jenkins2\workspace\GH_Tribler_PR_Tests\PR_async_win64\tribler\src\tribler-core\tribler_core\tests\tools\tools.py", line 12, in wrapper
    await wait_for(coro(*args, **kwargs), timeout)
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\asyncio\tasks.py", line 442, in wait_for
    return fut.result()
  File "C:\workspace\jenkins2\workspace\GH_Tribler_PR_Tests\PR_async_win64\tribler\src\tribler-core\tribler_core\modules\tunnel\test_full_session\test_tunnel_community.py", line 20, in test_anon_download
    await wait_for(download.future_finished, timeout=15)
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\asyncio\tasks.py", line 449, in wait_for
    raise futures.TimeoutError()
concurrent.futures._base.TimeoutError: 
```

**Comment by devos50** [comment]
Another one, might be related:

```
======================================================================
ERROR: Test whether we can successfully update a channel and download the new version
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\site-packages\asynctest\case.py", line 297, in run
    self._run_test_method(testMethod)
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\site-packages\asynctest\case.py", line 354, in _run_test_method
    self.loop.run_until_complete(result)
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\site-packages\asynctest\case.py", line 224, in wrapper
    return method(*args, **kwargs)
  File "c:\users\heteoma\appdata\local\programs\python\python37\lib\asyncio\base_events.py", line 579, in run_until_complete
    return future.result()
concurrent.futures._base.CancelledError: 
-------------------- >> begin captured logging << --------------------
```

**Comment by egbertbouman** [comment]
This first stacktrace is probably because downloading takes longer after we added the DHT check.

The second one seems unrelated. My guess is that it happens during download removal or shutdown in `download_channel` when `future_finished` is cancelled. The `CancelledError` needs to be handled properly, as ignoring it would trigger channel processing of a possibly removed/incomplete torrent.

I've added a fix for both in https://github.com/Tribler/tribler/pull/5405.
